controllers/charmedk8scontrolplane_controller.go

```diff
@@ -471,3 +546,136 @@ func (r *CharmedK8sControlPlaneReconciler) generateBootstrapConfig(ctx context.C
 
 	return bootstrapRef, nil
 }
+
+func (r *CharmedK8sControlPlaneReconciler) reconcileKubeconfig(ctx context.Context, cluster *clusterv1.Cluster, kcp *controlplanev1beta1.CharmedK8sControlPlane, machines []clusterv1.Machine, jujuClient *juju.Client, modelUUID string) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+	log.Info("reconciling kubeconfig")
+	endpoint := cluster.Spec.ControlPlaneEndpoint
+	if endpoint.IsZero() {
+		return ctrl.Result{}, nil
+	}
+
+	kubeConfigSecret := &kcore.Secret{}
+	kubeConfigSecret.Name = cluster.Name + "-kubeconfig"
+	kubeConfigSecret.Namespace = cluster.Namespace
+
+	err := r.Get(ctx, types.NamespacedName{Name: kubeConfigSecret.Name, Namespace: kubeConfigSecret.Namespace}, kubeConfigSecret)
+	if err != nil && apierrors.IsNotFound(err) {
```

Comment:
What if there's an error besides IsNotFound? Do we need to log and return it?

Reply:
added a codepath for when the error is not a not found, in that case it is logged and returned